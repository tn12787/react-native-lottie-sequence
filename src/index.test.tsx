import React from 'react';
import { shallow } from 'enzyme';
import Lottie from 'lottie-react-native';

import LottieSequence from './index';

const testSource = {
  v: '5.6.6',
  fr: 60,
  ip: 0,
  op: 297,
  w: 2200,
  h: 2200,
  nm: 'Welcome',
  ddd: 0,
  assets: [],
  layers: [],
};

describe('LottieSequence', () => {
  describe('Rendering', () => {
    it('should match visible snapshot', () => {
      const wrapper = shallow(<LottieSequence sources={[testSource]} />);
      expect(wrapper).toMatchSnapshot('LottieSequence Snapshot');
    });
  });

  describe('Contents', () => {
    it('should render a Lottie animation with sources + props provided', () => {
      const wrapper = shallow(<LottieSequence sources={[testSource]} />);

      const renderedAnimation = wrapper.find(Lottie);
      expect(renderedAnimation).toHaveLength(1);
    });
  });
});
