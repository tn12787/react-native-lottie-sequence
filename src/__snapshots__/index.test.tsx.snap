// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LottieSequence Rendering should match visible snapshot: LottieSequence Snapshot 1`] = `
<LottieView
  autoPlay={true}
  autoSize={false}
  enableMergePathsAndroidForKitKatAndAbove={false}
  loop={false}
  onAnimationFinish={[Function]}
  progress={0}
  resizeMode="contain"
  source={
    Object {
      "assets": Array [],
      "ddd": 0,
      "fr": 60,
      "h": 2200,
      "ip": 0,
      "layers": Array [],
      "nm": "Welcome",
      "op": 297,
      "v": "5.6.6",
      "w": 2200,
    }
  }
  speed={1}
/>
`;
